src/backends/quantumoptics/quantumoptics.jl

```diff
@@ -37,11 +37,13 @@ function observable(state::Union{<:Ket,<:Operator}, indices, operation)
     expect(op, state)
 end
 
-function project_traceout!(state::Union{Ket,Operator},stateindex,psis::Vector{<:Ket})
+function project_traceout!(state::Union{Ket,Operator},stateindex,psis::Base.AbstractVecOrTuple{Ket})
```

Comment:
So, the problem is that the symbolic basis states like `Z1`, `X2` etc aren't direct subtypes of `Ket`, but `SpecialKet` [here](https://github.com/QuantumSavory/QuantumSymbolics.jl/blob/b95474d90664259800a132d30241c4560f27d028/src/QSymbolicsBase/predefined.jl#L5). 
Should I just replace it with 
```julia
function project_traceout!(state::Union{Ket,Operator},stateindex,psis::Base.AbstractVecOrTuple{<:Union{Ket, QuantumSymbolics.SpecialKet}})
```